src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -758,7 +760,7 @@ export class ExposureNotificationService {
   private async recordKeySubmission() {
     const currentStatus = this.exposureStatus.get();
     if (currentStatus.type !== ExposureStatusType.Diagnosed) return;
-    this.exposureStatus.append({needsSubmission: false, submissionLastCompletedAt: getCurrentDate().getTime()});
+    this.exposureStatus.append({needsSubmission: false, hasShared: true, submissionLastCompletedAt: getCurrentDate().getTime()});
```

Comment:
Per the other comment if you force this to false for this PR we could merge without breaking existing functionality :) and you could keep the other logic.